File: Java/ExtremeTournament/src/extreme/model/Commentaire.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package extreme.model;

import java.sql.Date;

/**
 *
 * @author ASUS
 */
public class Commentaire {
    private int id_commentaire ;
    private String text ;
    private Date date_comment ; 
    private int nbr_reports ; 
    private int id_publication;
    private int id_user;

    public Commentaire(int id_commentaire, String text, Date date_comment) {
        this.id_commentaire = id_commentaire;
        this.text = text;
        this.date_comment = date_comment;
    }

    public Commentaire(String text, Date date_comment, int id_user) {
        this.text = text;
        this.date_comment = date_comment;
        this.id_user= id_user;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }
    
    

    public Commentaire(String text) {
        this.text = text;
    }
    
    

    public Commentaire() {
    }

    public Commentaire(int id_commentaire, String text, Date date_comment,  int id_publication,int nbr_reports) {
        this.id_commentaire = id_commentaire;
        this.text = text;
        this.date_comment = date_comment;
        this.nbr_reports = nbr_reports;
        this.id_publication = id_publication;
    }

    public int getNbr_reports() {
        return nbr_reports;
    }

    public void setNbr_reports(int nbr_reports) {
        this.nbr_reports = nbr_reports;
    }

    public int getId_publication() {
        return id_publication;
    }

    public void setId_publication(int id_publication) {
        this.id_publication = id_publication;
    }

    
    

    public int getId_commentaire() {
        return id_commentaire;
    }

    public void setId_commentaire(int id_commentaire) {
        this.id_commentaire = id_commentaire;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Date getDate_comment() {
        return date_comment;
    }

    public void setDate_comment(Date date_comment) {
        this.date_comment = date_comment;
    }

    @Override
    public String toString() {
        return "Commentaire{" + "id_commentaire=" + id_commentaire + ", text=" + text + ", date_comment=" + date_comment + '}';
    }
    
 
    
    
    
            
    
}

File: ExtremeTournament/src/Entity/Commentaire.php
<?php

namespace App\Entity;

use App\Repository\CommentaireRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use phpDocumentor\Reflection\PseudoTypes\IntegerRange;
use phpDocumentor\Reflection\Types\Integer;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * @ORM\Entity(repositoryClass=CommentaireRepository::class)
 */
class Commentaire
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id_commentaire;

    /**
     * @Assert\NotBlank(message="Text doit etre non vide")
     * @ORM\Column(type="string", length=150)
     */
    private $text;

    /**
     * @ORM\Column(type="date")
     */
    public $date_comment;

    /**
     * @ORM\ManyToOne(targetEntity=Publication::class, inversedBy="commentaires")
     * @ORM\JoinColumn(name="id_publication",referencedColumnName="id_publication",nullable=false)
     */
    public $id_publication;

    /**
     * @ORM\Column(type="integer",)
     */
    private $nbr_reports=0;

    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="comments")
     * @ORM\JoinColumn(name="id_user",referencedColumnName="id_user",nullable=false)
     */
    public $id_user;


    public function __construct()
    {
        $this->id_publication = new ArrayCollection();
        $this->id_user = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id_commentaire;
    }

    public function getText(): ?string
    {
        return $this->text;
    }

    public function setText(string $text): self
    {
        $this->text = $text;

        return $this;
    }

    public function getDateComment(): ?\DateTimeInterface
    {
        return $this->date_comment;
    }

    public function setDateComment(\DateTimeInterface $date_comment): self
    {
        $this->date_comment = $date_comment;

        return $this;
    }
    public function setidpublication(?Publication $id_publication): self
    {
        $this->id_publication = $id_publication;

        return $this;
    }

    /**
     * @return Collection<int, publication>
     */
    public function getIdPublication():?Collection    /*publication*/
    {
        return $this->id_publication;
    }


    public function addIdPublication(publication $idPublication): self
    {
        if (!$this->id_publication->contains($idPublication)) {
            $this->id_publication[] = $idPublication;
        }

        return $this;
    }

    public function removeIdPublication(publication $idPublication): self
    {
        $this->id_publication->removeElement($idPublication);

        return $this;
    }

    public function getNbrReports(): ?int
    {
        return $this->nbr_reports;
    }

    public function setNbrReports(int $nbr_reports): self
    {
        $this->nbr_reports = $nbr_reports;

        return $this;
    }

    public function getIdUser(): ?Collection     /*user*/
    {
        return $this->id_user;
    }

    public function setIdUser(?user $id_user): self
    {
        $this->id_user = $id_user;

        return $this;
    }

    public function addIdUser(user $idUser): self
    {
        if (!$this->id_user->contains($idUser)) {
            $this->id_user[] = $idUser;
        }

        return $this;
    }

    public function removeIdUser(user $idUser): self
    {
        $this->id_user->removeElement($idUser);

        return $this;
    }


    public function __toString()
    {
        return (string) $this->id_user;
    }

}
